com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/validators/parameters/RoutesValidator.java

```diff
@@ -0,0 +1,118 @@
+package com.sap.cloud.lm.sl.cf.core.validators.parameters;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import com.sap.cloud.lm.sl.cf.core.message.Messages;
+import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
+import com.sap.cloud.lm.sl.common.SLException;
+import com.sap.cloud.lm.sl.mta.model.v1.Module;
+
+public class RoutesValidator implements ParameterValidator {
+
+    private final Map<String, ParameterValidator> validators;
+
+    public RoutesValidator() {
+        this.validators = new HashMap<String, ParameterValidator>();
+        ParameterValidator routeValidator = new RouteValidator();
+        this.validators.put(routeValidator.getParameterName(), routeValidator);
+    }
+
+    @Override
+    public boolean isValid(Object routes) {
+        List<Map<String, Object>> routesList = applyRoutesType(routes);
+
+        if (routesList == null) {
+            return false;
+        }
+
+        for (Map<String, Object> routesElem : routesList) {
+            boolean elemIsValid = routesElem.keySet()
+                .stream()
+                .map(key -> this.validators.get(key) != null && this.validators.get(key)
+                    .isValid(routesElem.get(key)))
+                .reduce(true, Boolean::logicalAnd);
+            if (!elemIsValid)
+                return false;
+        }
+
+        return true;
+    }
+
+    public Object attemptToCorrect(Object routes) {
+        List<Map<String, Object>> routesList = applyRoutesType(routes);
+
+        if (routesList == null) {
+            throw new SLException(Messages.COULD_NOT_PARSE_ROUTE);
+        }
+
+        List<Map<String, Object>> correcteRoutesList = new ArrayList<Map<String, Object>>();
+
+        routesList.forEach(elem -> correcteRoutesList.add(attemptToCorrectParameterMap(elem)));
```

Comment:
Change to :
routesList.stream().map(route -> attemptToCorrectParameterMap(route)).collect(Collectors.toList());